data/scripts/actions/quests/threatened_dreams/lever.lua

```diff
@@ -1,94 +1,95 @@
 	},
-	teleportPosition = Position(33617, 32567, 13),
 	bossPosition = Position(33617, 32561, 13),
-	specPos = Position(33618, 32523, 15)
+	specPos = {
+		from = Position(33607, 32553, 13),
+		to = Position(33627, 32570, 13)
+	},
+	exit = Position(33618, 32523, 15),
+	storage = Storage.ThreatenedDreams.FacelessBaneTime
 }
 
 local threatenedLever = Action()
+
 function threatenedLever.onUse(player, item, fromPosition, target, toPosition, isHotkey)
-	if item.itemid == config.leverId then
-		-- Check if the player that pulled the lever is on the correct position
-		if player:getPosition() ~= config.playerPositions[1] then
-			player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You can\'t start the battle.")
-			return true
-		end
+	if config.playerPositions[1].pos ~= player:getPosition() then
+		return false
+	end
 
-		local team, participant = {}
-		for i = 1, #config.playerPositions do
-			participant = Tile(config.playerPositions[i]):getTopCreature()
+	local spec = Spectators()
+	spec:setOnlyPlayer(false)
+	spec:setRemoveDestination(config.exit)
+	spec:setCheckPosition(config.specPos)
+	spec:check()
 
-			-- Check there is a participant player
-			if participant and participant:isPlayer() then
-				-- Check participant level
-				if participant:getLevel() < config.requiredLevel then
-					player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "All the players need to be level ".. config.requiredLevel .." or higher.")
-					return true
-				end
+	if spec:getPlayers() > 0 then
+		player:say("There's someone fighting with Faceless Bane.", TALKTYPE_MONSTER_SAY)
+		return true
+	end
 
-				-- Check participant boss timer
-				if config.daily and participant:getStorageValue(Storage.ThreatenedDreams.FacelessBaneTime) > os.time() then
-					player:getPosition():sendMagicEffect(CONST_ME_POFF)
-					player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You or a member in your team have to wait ".. config.timeToFightAgain .."  hours to face Faceless Bane again!")
-					return true
-				end
-				team[#team + 1] = participant
-			end
+	local lever = Lever()
+	lever:setPositions(config.playerPositions)
+	lever:setCondition(function(creature)
+		if not creature or not creature:isPlayer() then
+			return true
 		end
-
-		-- Check if a team currently inside the boss room
-		local specs, spec = Game.getSpectators(config.centerRoom, false, false, 14, 14, 13, 13)
-		for i = 1, #specs do
-			spec = specs[i]
-			if spec:isPlayer() then
-				player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "There's someone fighting with Faceless Bane.")
-				return true
+		if creature:getLevel() < config.requiredLevel then
+			creature:sendTextMessage(MESSAGE_EVENT_ADVANCE, "All the players need to be level " .. config.requiredLevel .. " or higher.")
+			return false
+		end
+		if creature:getStorageValue(config.storage) > os.time() then
+			local info = lever:getInfoPositions()
+			creature:getPosition():sendMagicEffect(CONST_ME_POFF)
```

Comment:
This logic should send const_me_poff to all the players at the lever that are not able to do it(this way is just sending to 1 player at a time.
I fixed that below, but i don't know if it's the best way to do this:

```
if creature:getStorageValue(config.storage) > os.time() then
			local info = lever:getInfoPositions()
			for _, v in pairs(info) do
				local newPlayer = v.creature
				if newPlayer then
					newPlayer:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You or a member in your team have to wait " .. config.timeToFightAgain .. " hours to face Faceless Bane again!")
					if newPlayer:getStorageValue(config.storage) > os.time() then
						newPlayer:getPosition():sendMagicEffect(CONST_ME_POFF)
					end
				end
			end
			return false
		end
```

Reply:
On tibia global only the first player (who can push the lever) receive something (msg or effect)

Reply:
Sure, what i'm suggesting here is an enhancement. In tibia global, storage is checked at the teleport/door. As on OTServers, many adms have custom teleports room to take players directly to the bosses lever, that's why i think this enhancement would be useful.
In case of king zelos for example, lever is for 10 players, this enhancement is for everyone at the lever know which player doesn't have cooldown